paasta_tools/paasta_cli/cmds/rollback.py

```diff
@@ -53,22 +54,47 @@ def add_subparser(subparsers):
     list_parser.set_defaults(command=paasta_rollback)
 
 
+def validate_given_instances(service, args_instances):
+    """Trying to find which instances are actually valid for the given service"""
+    service_instances = list_all_instances_for_service(service)
+    invalid_instances = None
+
+    if not args_instances:
+        valid_instances = service_instances
+    else:
+        args_instances = args_instances.split(",")
+        valid_instances = set(args_instances).intersection(service_instances)
+        invalid_instances = set(args_instances).difference(service_instances)
+
+    return valid_instances, invalid_instances
+
+
 def paasta_rollback(args):
     """Call mark_for_deployment with rollback parameters"""
     service = figure_out_service_name(args)
     cluster = args.cluster
-    instance = args.instance
     git_url = get_git_url(service)
     commit = args.commit
 
     if cluster in list_clusters(service):
-        returncode = mark_for_deployment(
-            git_url=git_url,
-            cluster=cluster,
-            instance=instance,
-            service=service,
-            commit=commit
-        )
+        instances, invalid = validate_given_instances(service, args.instance)
+
+        if invalid:
+            print "WARNING: These instances are not valid and will not be deployed: %s.\n" % (",").join(invalid)
+
+        if instances:
+            for instance in instances:
+                print "Deploying %s to %s.%s.\n" % (service, cluster, instance)
```

Comment:
This print line is kinda misleading. Can you remove it and let the normal stdout of the mark_for_deployment function show through?